packages/@lwc/synthetic-shadow/src/shared/node-collection-util.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { ArrayConstructor } from '@lwc/shared';
+/**
+ * Issue #1545 - Writing a custom util to convert NodeList and HTMLCollection instances to array.
+ * This is to aviod conflict with some legacy third party libraries like prototype.js
+ * that patch Array.prototype and are incapable of handling NodeList and HTMLCollection.
+ */
+export function arrayFromCollection<T extends Node, K extends Element>(
```

Comment:
can we have two methods, one for htmlcollection and one for nodelist? even if the underlying implementation is the same... I feel that this PR is messing with the types.